packages/search-in-workspace/src/browser/search-in-workspace-widget.tsx

```diff
@@ -57,6 +58,9 @@ export class SearchInWorkspaceWidget extends BaseWidget implements StatefulWidge
     protected resultContainer: HTMLElement;
 
     @inject(SearchInWorkspaceResultTreeWidget) protected readonly resultTreeWidget: SearchInWorkspaceResultTreeWidget;
+    @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService;
+
+    protected isWorkspacePresent: boolean = false;
```

Comment:
don't keep the duplicate state in the UI component, it will get out of sync with `WorkspaceService` state

How about replacing it with a function which checks `WorkspaceService` state each time?

Reply:
thanks @akosyakov, I initially had it the way you suggested and I think its necessary in this case to check the state each time to keep the sync.